### DISCUSSION TITLE
Nvidia 10xx Series - optimizations for FLux/ SDXL SD 1.5 - Correct Flags

### DISCUSSION THREAD

**Discussion by motolo** [comfyanonymous/ComfyUI#4604] [General]
Latest versions of xformers have issues with FP16 which help build speed. 
If Xformers is not used with the --force-fp32 flag it does not work, in forge they have introduced a fix for this problem.

https://www.reddit.com/r/StableDiffusion/comments/1exw7kw/forge_fix_for_nvidia_10xx_gpus_2x_faster/

Having said that I tested all the various flags to get the most out of our old card

For Flux/SDXL this is the best flag that improves times, not saturating the vram:

--disable-xformers --use-split-cross-attention --cache-classic --lowvram

However, this flag slows down SD1.5 considerably

For SD1.5/SDXL use this Flag:

--force-fp32 --cache-classic --lowvram

These are the settings that I think are best.

It's a shame though that there isn't a single way to have complete optimization.

**Comment by JorgeR81** [comment]
- GTX 1070 
- 32 GB RAM
- Windows 10
- pytorch version: 2.3.1+cu121
- no xformers installed

I tried `--use-split-cross-attention --cache-classic --lowvram`, but Flux seems to be slightly slower.

With these flags, less VRAM is used ( 5.8 GB, vs 7.3 GB ), which seems to be unnecessarily low. 
So that could be slowing down generation a bit.  

But these could be useful in case I ever get an OOM error.

---

By the way, I just realized I don't need `--force-fp16` anymore. 
This prevented an error with IPAdapter FaceID SDXL, but it seems it's not needed anymore !

**Reply by motolo** [reply]
It's amazing how each configuration needs a different flag. I will then put the images of the different results. 

Anyway I'm on:

Linux Arch
Pytorch 2.3.1cu121
32GB RAM
Nvidia 1060 6gb vram

Remember to update comfyui to the latest version.

P. S. problem with Face ID SDXL has been corrected in fact it is no longer needed FP16

**Reply by motolo** [reply]
Which version of flux do you use? I use  Schnell fp8 all included.

**Reply by JorgeR81** [reply]
I use Flux dev.  With 12 or 15 steps I can get decent results.
There are many many more loras for dev.
I hope they can also make "turbo" lora for dev.

Also, I use the Q8_0 and Q4_K_S versions.  
For me, they have about the same speed as FP16 and FP8 respectively, but they load faster.
https://github.com/city96/ComfyUI-GGUF
